app/file.go

```diff
@@ -1129,6 +1129,15 @@ func (a *App) generateMiniPreview(fi *model.FileInfo) {
 			mlog.Debug("generateMiniPreview: prepareImage failed", mlog.Err(imgErr),
 				mlog.String("fileinfo_id", fi.Id), mlog.String("channel_id", fi.ChannelId),
 				mlog.String("creator_id", fi.CreatorId))
+
+			// Since this file is not a valid image (for whatever reason), prevent this fileInfo
+			// from entering generateMiniPreview in the future
+			fi.UpdateAt = model.GetMillis()
+			fi.MimeType = "invalid-" + fi.MimeType
+			if _, err2 := a.Srv().Store.FileInfo().Upsert(fi); err2 != nil {
```

Comment:
err2 because err above is *model.AppError

Reply:
Hah, I should have waited for this comment :p